src/sql-parser/src/ast/defs/query.rs

```diff
@@ -28,6 +28,19 @@ use crate::ast::{Expr, FunctionArgs, Ident, ObjectName, SqlOption};
 
 // This represents the metadata that lives next to an AST, as we take it through
 // various stages in the planning process.
+//
+// Conceptually, when we first receive an AST from the parsing process, it only
```

Comment:
I think this makes sense! So will each new piece of semantic information we want to store be added to `AstInfo` as an additional `type`?

Reply:
Also, not in this PR but related to `AstInfo`, is there a reason why we'd want the table references (`type Table`) to be a separate table from other types of references (`type View` or `type Type`)?

Reply:
Yes to question 1, I think anything that needs to vary between the two phases needs to be an associated type on `AstInfo`. Re: the second, I'm not sure! If types are also associated with `GlobalIDs` then they could probably be the same thing, which would be nice (but will warrant a rename, I guess)!